{
  try {
    runnable.run();
  }
 catch (  Exception e) {
    assertSame(ecls,e.getClass());
    LOG.info("got expected: " + e);
    return;
  }
  assertTrue("Should've thrown " + ecls.getSimpleName(),false);
}
